stdlib/runtime/wasi.gr

```diff
@@ -161,3 +164,97 @@ let _ETIMEDOUT = 73n
 let _ETXTBSY = 74n
 let _EXDEV = 75n
 let _ENOTCAPABLE = 76n
+
+exception SystemError(Number)
+
+let stringOfSystemError = (code) => {
+  let n = WasmI32.shrS(WasmI32.fromGrain(code), 1n)
+  let (==) = WasmI32.eq
+  match (n) {
+    n when n == _ESUCCESS => "SystemError: No error occurred. System call completed successfully.",
```

Comment:
This one feels weird. Maybe it's just the message?

Reply:
I think it's because these are error messages and this one isn't an error 😛 

The messages all come from the official spec: https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#-errno-variant

Grain currently doesn't ever throw a `SystemError` if it's `ESUCCESS`, but I left it in here for completeness.